daft/expressions/expressions.py

```diff
@@ -196,6 +196,63 @@ def list_(*items: Expression | str):
     return Expression._from_pyexpr(native.list_([col(i)._expr if isinstance(i, str) else i._expr for i in items]))
 
 
+def struct(*fields: Expression | str) -> Expression:
+    """Constructs a struct from the input field expressions.
+
+    Example:
+        >>> import daft
+        >>> from daft import col
+        >>> df = daft.from_pydict({"a": [1, 2, 3], "b": ["a", "b", "c"]})
+        >>> df.select(daft.struct(col("a") * 2, col("b"))).show()
+        ╭───────────────────────────╮
+        │ struct                    │
+        │ ---                       │
+        │ Struct[a: Int64, b: Utf8] │
+        ╞═══════════════════════════╡
+        │ {a: 2,                    │
+        │ b: a,                     │
+        │ }                         │
+        ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+        │ {a: 4,                    │
+        │ b: b,                     │
+        │ }                         │
+        ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+        │ {a: 6,                    │
+        │ b: c,                     │
+        │ }                         │
+        ╰───────────────────────────╯
+        <BLANKLINE>
+        (Showing first 3 of 3 rows)
+
+    Args:
+        inputs: Expressions to be converted into struct fields.
+
+    Returns:
+        An expression for a struct column with the input columns as its fields.
+    """
+    pyinputs = []
+    for field in fields:
+        if isinstance(field, Expression):
+            pyinputs.append(field._expr)
+        elif isinstance(field, str):
+            pyinputs.append(col(field)._expr)
+        else:
+            raise TypeError("expected Expression or str as input for struct()")
+    return Expression._from_pyexpr(_struct(pyinputs))
+
+
+def to_struct(*fields: Expression | str) -> Expression:
```

Comment:
This should probably remain a classmethod on Expression for backward compatibility?